### PR TITLE
Suspend Yandex provider due to protobuf limitation

### PR DESCRIPTION
**Pull request by potiuk**
Yandex provider brings protobuf dependency down to <4 and we are gearing up to updating it everywhere else. Protobuf3 support ends in Q2 2023 for Python https://protobuf.dev/support/version-support/#python

Yandex is the last provider that we do not closely collaborate with on fixing it

* Gogle provider dependencies are actively upgraded to latest version by Google led team: #30067 (some of the libraries are already updated) with target to update all dependencies by mid-May
* Apache-Beam has already merged protobuf4 support https://github.com/apache/beam/pull/25874 with the target of relasing it in 2.47.0 mid-May
* The mysql-connector-python in MySQL provider is already turned into optional dependency: #30487

The only remaining dependency limiting us to protobuf 3 (<3.21) is yandexcloud. We've opened an issue to yandexcloud
https://github.com/yandex-cloud/python-sdk/issues/71 3 weeks ago and while there was an initial interest, there is no progress on the issue, therefore - in order to prepare for running all the tests and final migration to protobuf4 we need to suspend Yandex provider - following the suspension process we agreed and got a LAZY CONSENSUS on in
the https://lists.apache.org/thread/g8b3k028qhzgw6c3yz4jvmlc67kcr9hj mailing list discussion.

The yandex provider can be removed from suspension by a PR reverting this change once yandexcloud dependency removes the protobuf limitation in their release and PR reverting this change (and fixing all tests and static check that will be needed) is the way it can be done.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

File: airflow/providers/yandex/provider.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ package-name: apache-airflow-providers-yandex
 name: Yandex
 description: |
     Yandex including `Yandex.Cloud <https://cloud.yandex.com/>`__
-suspended: false
+suspended: true
```
Comment (eladkal) [inline airflow/providers/yandex/provider.yaml]:
```suggestion
suspended: true # see https://github.com/apache/airflow/pull/30667/
```

Comment (potiuk) [reply airflow/providers/yandex/provider.yaml]: Yep.

**Review by hussein-awala — APPROVED** [review]
wow, it shows how much #30422 has simplified the suspension process! LGTM

**Comment by potiuk** [comment]
> wow, it shows how much #30422 has simplified the suspension process! LGTM

Yep. All with a single flag flipped. That's the beauty of  having everything under control of our CI/Breeze/pre-commits framework.

**Pull request merged**